scanpy/tests/blackdiff.py

```diff
@@ -45,8 +45,11 @@ def black_diff(src: Path, differ=Differ()) -> Tuple[int, int]:
     ]
     for file, (added, removed) in sorted(excluded, key=lambda sd: -sum(sd[1])):
         print(f'{file}: +{added} -{removed}')
-        if added + removed < thresh:
-            print('File has < 10 changes to black formatting. Do it!')
+        if 0 < added + removed < thresh:
+            print(
+                f'File: {file}: has < {threshold} changes to '
+                f'black formatting. Do it!'
+            )
```

Comment:
No, if it’s 0, it should be removed from `tool.black.exclude` in `pyproject.toml`. Good idea to change the message in that case though. It should be:

```py
if added + removed == thresh:
    print('This file has 0 changes to black formatting, good job! Please remove it from “tool.black.exclude”')
elif added + removed < thresh:
    print('This file has < {thresh} changes to black formatting. Do it!')
```

Also no need to repeat the file name, it’s the one from the line above.

Reply:
Added it because it was not obvious. I didn't know what was happening until I dig in the code and realized that the file before was the one throwing the error. At first I thought the code checks all files and then gives a report.